mock/mock.go

```diff
@@ -833,6 +833,10 @@ type IsTypeArgument struct {
 // For example:
 //
 //	args.Assert(t, IsType(""), IsType(0))
+//
+// Note: IsType checks for the exact type you give it.
+// If you pass a nil interface, it has no type and will never match.
+// To match an interface value, pass a non-nil value of the concrete type you expect.
```

Comment:
This isn't quite accurate, the limitation is that you cannot pass an interface. You actually could match a `<nil>` type. When you call IsType you're actually asking Go to convert your type to interface{} (which it may already be) and then call IsType with it. This means any interface type will be lost when making this call. IsType then looks at the concrete type inside the interface{} it holds. Consider these examples:
```go
func IsType(t interface{}) {
	fmt.Printf("%T\n", t)
}

...

i := 1
IsType(i) // int

var a any
IsType(a) // <nil>

a = i
IsType(a) // int

var r io.Reader
var err error
r, err = os.Create("file")
if err != nil {
	panic(err)
}
IsType(r) // *os.File

IsType(&r) // *io.Reader (pointer to interface is a concrete type)
```

Perhaps a better phrasing?
```suggestion
//
// Mock cannot match interface types because the contained type will be  passed
// to both IsType and Mock.Called, for the zero value of all interfaces this
// will be <nil> type.
```